### PR TITLE
feat(ui): add query builder

### PR DESCRIPTION
**Pull request by misi-bp**
## Description

The query builder can define the scope (~data filter), and it also can be modified manually.
Weakness: It's not able to create lambda functions (RHS widgets are not properly customizable).
The manual editor, however, can be used for this purpose.
The reason why the RHS is not so customizable: https://github.com/ukrbublik/react-awesome-query-builder/issues/109
The solution's link is broken, no documentation for it, and my experiments failed. The bottlenecks, when speaking of field/value types and widgets are as follows:
- A custom widget should be used, but only a single value-widget can be used with one selected property type.
- The operator dropdown should be reusable in our custom component, but it isn't.

Tickets:
https://cisco-eti.atlassian.net/browse/PA-14842
https://cisco-eti.atlassian.net/browse/PA-15020
https://cisco-eti.atlassian.net/browse/PA-15341
https://cisco-eti.atlassian.net/browse/PA-15342

Notes:
* react-app-rewired was necessary to override the Webpack config under the react-scripts (CRA, create-react-app).
* override the webpack config was necessary to replace the previously existing packages of node.
* This PR also contains the frontend part of how the saved query to the BE can be opened. Explanation: After a thorough understanding of the documentation, it turned out that this process is done by saving not just the query, but the configuration as well to the BE. This weird solution can ensure - or at least that's my intention -, a saved query can be opened anytime with its own config settings (properties/fields), but every change to it is done based on the new config settings.

My suggestion:
1. Let's get rid of this package. It's not maintained properly, and its codebase is very 'oldschool' (class components, poor type checks, bug when drag&dropping rule groups (it's prevented now), very rigid rule structures.
2. Once we've upgraded the app with a UI kit, AND considering the fact we already have the conventional functions for fields, the types, operators listed, we can create our own querybuilder, that should be flexible enough.

## Type of Change

[ ] Bug Fix  
[X] New Feature  
[ ] Breaking Change  
[ ] Refactor  
[ ] Documentation  
[ ] Other (please describe)  

## Checklist

- [X] I have read the [contributing guidelines](https://github.com/openclarity/vmclarity/blob/main/CONTRIBUTING.md)
- [X] Existing issues have been referenced (where applicable)
- [X] I have verified this change is not present in other open pull requests
- [X] Functionality is documented
- [X] All code style checks pass
- [ ] New code contribution is covered by automated tests
- [ ] All new and existing tests pass

### CODE REVIEW

File: ui/src/layout/Scans/ScanConfigWizardModal/CustomQueryBuilder/CustomQueryBuilder.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    const [{ loading, data, error }] = useFetch(`${isDev ? "http://localhost:3000" : ""}/api/openapi.json`, { isAbsoluteUrl: true });
+    const [config, setConfig] = useState(BASIC_CONFIG);
+    const [queryState, setQueryState] = useState({
+        config,
+        tree: QbUtils.checkTree(QbUtils.loadTree(queryValue), config),
+    });
+
+    const [, , helpers] = useField(name); //const [field, , helpers] 
+    const { setValue } = helpers;
+    //const { value } = field;
```
Comment (gallotamas) [inline ui/src/layout/Scans/ScanConfigWizardModal/CustomQueryBuilder/CustomQueryBuilder.js:36]: please remove unused code

File: ui/src/layout/Scans/ScanConfigWizardModal/CustomQueryBuilder/widgets/LengthWidget.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
+import React, { useCallback } from "react";
+//import { AntdWidgets } from '@react-awesome-query-builder/antd';
+import { InputNumber, Col } from "antd";
+
+// const {
+//     FieldSelect,
+//     TextWidget,
+//     NumberWidget,
+// } = AntdWidgets;
```
Comment (gallotamas) [inline ui/src/layout/Scans/ScanConfigWizardModal/CustomQueryBuilder/widgets/LengthWidget.js:9]: please remove unused code

File: ui/src/layout/Scans/ScanConfigWizardModal/CustomQueryBuilder/widgets/LengthWidget.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+    //const operator = value && Array.isArray(value) && value[0];
+    const numValue = value && Array.isArray(value) && value[1];
+
+    /*
+    const handleChangeOperator = useCallback((operatorvalue) => {
+        if (value && Array.isArray(value)) {
+            setValue(prevVal => ([operatorvalue, prevVal[1]]));
+        }
+    }, [setValue, value]);
```
Comment (gallotamas) [inline ui/src/layout/Scans/ScanConfigWizardModal/CustomQueryBuilder/widgets/LengthWidget.js:39]: please remove unused code

File: ui/src/layout/Scans/ScanConfigWizardModal/CustomQueryBuilder/CustomQueryBuilder.scss [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+      border-inline-end-color: rgba(128, 128, 128, 0.1);
+    }
+  }
+}
+
+.group--actions.group--actions--tr {
+
+  display: flex;
+  flex-direction: row-reverse;
+  justify-content: end;
```
Comment (gallotamas) [inline ui/src/layout/Scans/ScanConfigWizardModal/CustomQueryBuilder/CustomQueryBuilder.scss:218]: let's use `flex-end`
I'm getting the following warning with `end`
`autoprefixer: end value has mixed support, consider using flex-end instead`

File: ui/src/layout/Scans/ScanConfigWizardModal/CustomQueryBuilder/CustomQueryBuilder.js (hunk 1 above)
Comment (misi-bp) [reply ui/src/layout/Scans/ScanConfigWizardModal/CustomQueryBuilder/CustomQueryBuilder.js]: done

File: ui/src/layout/Scans/ScanConfigWizardModal/CustomQueryBuilder/CustomQueryBuilder.js [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+        </div>
+    ), [queryState.tree.size]);
+
+    const updateResult = throttle((immutableTree, config) => {
+        setQueryState(prevState => ({ ...prevState, tree: immutableTree, config }));
+        // eslint-disable-next-line
+    }, 100);
+
+    const onChange = useCallback((immutableTree, config) => {
+        //const jsonTree = QbUtils.getTree(immutableTree);
```
[the remark this answers is not in the document]
Comment (misi-bp) [reply ui/src/layout/Scans/ScanConfigWizardModal/CustomQueryBuilder/CustomQueryBuilder.js]: done

File: ui/src/layout/Scans/ScanConfigWizardModal/CustomQueryBuilder/widgets/LengthWidget.js (hunk 2 above)
Comment (misi-bp) [reply ui/src/layout/Scans/ScanConfigWizardModal/CustomQueryBuilder/widgets/LengthWidget.js]: very done

File: ui/src/layout/Scans/ScanConfigWizardModal/CustomQueryBuilder/widgets/LengthWidget.js (hunk 3 above)
Comment (misi-bp) [reply ui/src/layout/Scans/ScanConfigWizardModal/CustomQueryBuilder/widgets/LengthWidget.js]: d

File: ui/src/layout/Scans/ScanConfigWizardModal/CustomQueryBuilder/CustomQueryBuilder.scss (hunk 4 above)
Comment (misi-bp) [reply ui/src/layout/Scans/ScanConfigWizardModal/CustomQueryBuilder/CustomQueryBuilder.scss:218]: d

File: ui/src/layout/Scans/ScanConfigWizardModal/CustomQueryBuilder/CustomQueryBuilder.js [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+
+// You can load query value from your backend storage (for saving see `Query.onChange()`)
+const queryValue = { id: QbUtils.uuid(), type: "group" };
+
+const CustomQueryBuilder = ({
+    initialQuery,
+    name
+}) => {
+    const isDev = process.env.NODE_ENV === "development";
+    const [{ loading, data, error }] = useFetch(`${isDev ? "http://localhost:3000" : ""}/api/openapi.json`, { isAbsoluteUrl: true });
```
[the remark this answers is not in the document]
Comment (misi-bp) [reply ui/src/layout/Scans/ScanConfigWizardModal/CustomQueryBuilder/CustomQueryBuilder.js:21]: Thanks, I was tired :)

[the remark this answers is not in the document]
Comment (misi-bp) [reply ui/src/layout/Scans/ScanConfigWizardModal/CustomQueryBuilder/CustomQueryBuilder.js:21]: done

File: ui/src/layout/Scans/ScanConfigWizardModal/CustomQueryBuilder/CustomQueryBuilder.scss (hunk 4 above)
Comment (misi-bp) [reply ui/src/layout/Scans/ScanConfigWizardModal/CustomQueryBuilder/CustomQueryBuilder.scss:218]: Chrome doesn't support 'end' yet, it looks like still Firefox and Opera are the only browsers to support it

Comment (gallotamas) [reply ui/src/layout/Scans/ScanConfigWizardModal/CustomQueryBuilder/CustomQueryBuilder.scss:218]: ok, so can we use `flex-end`?

Comment (misi-bp) [reply ui/src/layout/Scans/ScanConfigWizardModal/CustomQueryBuilder/CustomQueryBuilder.scss:218]: nope, it doesn't work in Chrome.

**Pull request closed without merging**